sklearn/metrics/cluster/_expected_mutual_info_fast.pyx

```diff
@@ -38,9 +38,10 @@ def expected_mutual_information(contingency, int n_samples):
     term1 = nijs / N
     # term2 is log((N*nij) / (a * b)) == log(N * nij) - log(a * b)
     # term2 uses the outer product
-    log_ab_outer = np.log(a)[:, np.newaxis] + np.log(b)
-    # term2 uses N * nij
-    log_Nnij = np.log(N * nijs)
+    log_a = np.log(a)
+    log_b = np.log(b)
```

Comment:
Slightly more memory efficient because we would not need to create the 2d array anymore.